modules/clightning/Makefile

```
@@ -0,0 +1,56 @@
+.PHONY: build-lightning
+
+
+CXX = clang++
+CXXFLAGS += -Wall -Wextra -std=c++20 -fPIC
+
+INCLUDE_DIRS += -I.
+INCLUDE_DIRS += -I../../include
+INCLUDE_DIRS += -I../../external/lightning
+INCLUDE_DIRS += -I../../external/lightning/ccan
+INCLUDE_DIRS += -I../../external/lightning/external/libwally-core/src/secp256k1/include
+
+CXXFLAGS += $(INCLUDE_DIRS)
+
+ROOT_DIR := $(realpath $(dir $(lastword $(MAKEFILE_LIST)))/../..)
+LIGHTNING_DIR := $(ROOT_DIR)/external/lightning
+CLIGHTNING_MAKEFILE := $(ROOT_DIR)/modules/clightning/Makefile-clightning
+CLIGHTNING_LIB_H := $(ROOT_DIR)/modules/clightning/clightning_lib.h
+CLIGHTNING_LIB_C := $(ROOT_DIR)/modules/clightning/clightning_lib.c
+LIBCLIGHTNING_WRAPPER := $(ROOT_DIR)/modules/clightning/libcln.a
+
+all: module.a
+
+$(LIGHTNING_DIR)/config.vars:
+	@echo "Running ./configure in $(LIGHTNING_DIR)..."
+	cd $(LIGHTNING_DIR) && ./configure
```

To maximize libFuzzer's effectiveness, we should build libcln.a with fuzzer instrumentation.  We *may* also want to use ASan like we do elsewhere.

```suggestion
	cd $(LIGHTNING_DIR) && ./configure --enable-address-sanitizer --enable-fuzzing --disable-valgrind CC=clang
```

---

Good idea!

Done